### ISSUE TITLE
taint2 relies on std::regex, which is buggy in gcc < 4.9

### ISSUE THREAD

**Issue by vigliag** [panda-re/panda#175]
Hi! While developing a plugin of mine, I found out I couldn't use std::regex, as it's [not supported by versions of GCC earlier than 4.9](http://stackoverflow.com/questions/12530406/is-gcc-4-8-or-earlier-buggy-about-regular-expressions) (on my ubuntu 14.04 there's 4.8.4), and I've resorted to using boost::regex instead.

Reading taint2's source, I've come across some lines which too make use of `std::regex` and `std::regex_match`
https://github.com/panda-re/panda/blob/master/panda/plugins/taint2/llvm_taint_lib.cpp#L1116
https://github.com/panda-re/panda/blob/master/panda/plugins/taint2/llvm_taint_lib.cpp#L1193
Unless I'm missing something in the makefile, I think this may also be a problem for taint2 (which seems to be built with gcc as well).

I'm able to observe a buggy behavior of `std::regex_match` on my ubuntu 14.04 with the following test code (shamelessy adapted from [this answer](http://stackoverflow.com/a/41186162) to use one of the aforementioned regexps): 

```
cat << EOF | g++ --std=c++11 -x c++ - && ./a.out
#include <regex>

#if __cplusplus >= 201103L &&                             \
    (!defined(__GLIBCXX__) || (__cplusplus >= 201402L) || \
        (defined(_GLIBCXX_REGEX_DFS_QUANTIFIERS_LIMIT) || \
         defined(_GLIBCXX_REGEX_STATE_LIMIT)           || \
             (defined(_GLIBCXX_RELEASE)                && \
             _GLIBCXX_RELEASE > 4)))
#define HAVE_WORKING_REGEX 1
#else
#define HAVE_WORKING_REGEX 0
#endif

#include <iostream>

int main() {
  std::string testString = "log";
  static const std::regex mathRegex(
    "sin|cos|tan|log|__isinf|__isnan|rint|floor|abs|fabs|ceil|exp2",
    std::regex::egrep);
  const auto result = std::regex_match(testString, mathRegex);

#if HAVE_WORKING_REGEX
  std::cerr << "<regex> works, look: " << std::boolalpha << result << std::endl;
#else
  std::cerr << "<regex> doesn't work, look: " << std::boolalpha << result << std::endl;
#endif
  return result ? EXIT_SUCCESS : EXIT_FAILURE;
}
EOF
```
note that the same code works if `testString` is "sin" or "cos".

**Comment by phulin** [comment]
Should be fixed by 9b0b341fc22f9198f5c3cc854758fefca570b2c7

**Comment by vigliag** [comment]
Great! Thanks!
